verification/src/http_server/handlers/verification/mod.rs

```diff
@@ -1,10 +1,71 @@
-use serde::Serialize;
+#![allow(dead_code)]
+
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
 
 pub mod routes;
 pub mod solidity;
 pub mod sourcify;
 
-#[derive(Debug, Serialize, PartialEq)]
+#[derive(Debug, Deserialize, Serialize, PartialEq)]
+pub struct ContractLibrary {
+    pub lib_name: String,
+    pub lib_address: String,
+}
+
+#[derive(Debug, Deserialize, Serialize, PartialEq)]
 pub struct VerificationResponse {
-    pub verified: bool,
+    pub message: String,
+    pub result: Option<VerificationResult>,
+    pub status: VerificationStatus,
+}
+
+#[derive(Debug, Deserialize, Serialize, PartialEq)]
+pub struct VerificationResult {
+    pub contract_name: String,
+    pub compiler_version: String,
+    pub evm_version: String,
+    pub constructor_arguments: Option<String>,
+    pub contract_libraries: Option<Vec<ContractLibrary>>,
+    pub abi: String,
+    pub sources: HashMap<String, String>,
+}
+
+#[derive(Debug, Deserialize, Serialize, PartialEq)]
+pub enum VerificationStatus {
+    #[serde(rename = "0")]
+    Ok,
+    #[serde(rename = "1")]
+    BytecodeDismatch,
+    #[serde(rename = "99")]
+    UnknowError,
+}
+
+impl VerificationStatus {
+    fn default_message(&self) -> &'static str {
+        match self {
+            Self::Ok => "OK",
+            Self::BytecodeDismatch => "Bytecode doesn't match, please try again",
+            Self::UnknowError => "Unknow error",
+        }
+    }
+}
+
+impl VerificationResponse {
+    pub fn ok(result: VerificationResult) -> Self {
+        Self {
+            message: VerificationStatus::Ok.default_message().into(),
+            result: Some(result),
+            status: VerificationStatus::Ok,
+        }
+    }
+
+    pub fn err(status: VerificationStatus, message: Option<String>) -> Self {
```

Comment:
may be make this as two separate functions
`fn err(status)` and `fn err_with_message(status, message)`?

Reply:
also may be accept `impl Display`, not string, so we could send actual errors here too, like `parse_something().map_err(|e| err(UnknownError, e)?;`

Reply:
I don't really like that we can send an error with `VerificationStatus::Ok`

If we decide to leave only `VerificationStatus::Ok` and `VerificationStatus::Error` options for now, it would be simpler as we will not need to pass the status in `err()` method.

Reply:
Also do we really need a default messages? What if `message` field would be empty in case if no messages are there at all (e.g. status is Ok)?